embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryService.kt

```diff
@@ -37,14 +39,20 @@
      */
     override fun sendSession(sessionMessage: SessionMessage, snapshotType: SessionSnapshotType) {
         val sanitizedSessionMessage = gatingService.gateSessionMessage(sessionMessage)
-        val sessionBytes = cacheManager.saveSession(sanitizedSessionMessage, snapshotType)
-        if (sessionBytes == null || snapshotType == SessionSnapshotType.PERIODIC_CACHE) {
+        cacheManager.saveSession(sanitizedSessionMessage, snapshotType)
+        if (snapshotType == SessionSnapshotType.PERIODIC_CACHE) {
             return
         }
 
         try {
-            val future = apiService.sendSession(sessionBytes) {
-                cacheManager.deleteSession(sessionMessage.session.sessionId)
+            val sessionId = sessionMessage.session.sessionId
+            val action = cacheManager.loadSessionAsAction(sessionId) ?: { stream ->
```

Comment:
This would only apply if the app had a session that was previously cached in the old format?

Reply:
This codepath is taken by all sessions. Sessions are written to disk in the cache before a HTTP request is sent. If the request fails that cache file is deleted after a 'retry' file is created in a separate directory.

The system is a little confusing but I would not want to change it unless we make wider changes to the persistence layer